lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -241,84 +288,92 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         }
 
         missing_fields = sorted(
-            [
-                field
-                for field in REQUIRED_INGRESS_RELATION_FIELDS
-                if ingress_data.get(field) is None
-            ]
+            field for field in REQUIRED_INGRESS_RELATION_FIELDS if ingress_data.get(field) is None
         )
 
         if missing_fields:
-            LOGGER.error(
-                "Missing required data fields for ingress relation: %s",
+            LOGGER.warning(
+                "Missing required data fields for %s relation: %s",
+                relation,
                 ", ".join(missing_fields),
             )
             self.model.unit.status = BlockedStatus(
-                f"Missing fields for ingress: {', '.join(missing_fields)}"
+                f"Missing fields for {relation}: {', '.join(missing_fields)}"
             )
 
-        # Conform to charm-relation-interfaces.
-        if "name" in ingress_data and "port" in ingress_data:
-            name = ingress_data["name"]
-            port = ingress_data["port"]
-        else:
-            name = ingress_data["service-name"]
-            port = ingress_data["service-port"]
-        event.relation.data[self.model.app]["url"] = f"http://{name}:{port}/"
+        if relation == "ingress":
+            # Conform to charm-relation-interfaces.
+            if "name" in ingress_data and "port" in ingress_data:
+                name = ingress_data["name"]
+                port = ingress_data["port"]
+            else:
+                name = ingress_data["service-name"]
+                port = ingress_data["service-port"]
+            event.relation.data[self.model.app]["url"] = f"http://{name}:{port}/"
+
+            # Create an event that our charm can use to decide it's okay to
+            # configure the ingress.
+            self.charm.on.ingress_available.emit()
+        elif relation == "ingress-proxy":
+            self.charm.on.ingress_proxy_available.emit()
 
-        # Create an event that our charm can use to decide it's okay to
-        # configure the ingress.
-        self.charm.on.ingress_available.emit()
 
-    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
-        """Handle a relation-broken event in the ingress relation."""
-        if not self.model.unit.is_leader():
-            return
+class IngressProvides(IngressBaseProvides):
+    """Class containing the functionality for the 'provides' side of the 'ingress' relation.
 
-        # Create an event that our charm can use to remove the ingress resource.
-        self.charm.on.ingress_broken.emit(event.relation)
+    Attrs:
+        charm: The charm in which the relation takes place.
```

Comment:
Can you specify which charm? Is this an instance of the ingress-integrator or the charm that is related to it?

Reply:
It is the charm that is related to it. I'll clarify :+1: